js/calendar.js

```diff
@@ -83,3 +91,153 @@ function refreshCalendar(nextState) {
+		var startYear = eventData.start.getFullYear()
+		var endYear = eventData.end.getFullYear();
+		const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+		var startDate = "";
+		var endDate = "";
+		if (startYear != endYear) {
+			// years differ - "dd MMM YYYY - dd MMM YYYY"
+			startDate = startDay + space + months[startMonth] + space + startYear;
+			endDate = endDay + space + months[endMonth] + space + endYear;
+			if (!allDay) {
+				// "dd MMM YYYY HH:MM - dd MMM YYYY HH:MM"
+				startDate += space + startHour + colon + startMinute;
+				endDate += space + startHour + colon + startMinute;
+			}
+		} else if (startMonth != endMonth) {
+			// same year, different month - "dd MMM - dd MMM"
+			startDate = startDay + space + months[startMonth];
+			endDate = endDay + space + months[endMonth];
+			if (!allDay) {
+				// "dd MMM HH:MM - dd MMM HH:MM"
+				startDate += space + startHour + colon + startMinute;
+				endDate += space + startHour + colon + startMinute;
+			}
+		} else if (startDay != endDay) {
+			// same year, same month, different day - "dd - dd MMM YYYY"
+			if (allDay) {
+				startDate = startDay;
+				endDate = endDay + space + months[endMonth] + space + endYear;
+			} else {
+				// "dd MMMM YYYY HH:MM - dd MMM YYYY HH:MM"
+				startDate = startDay + space + months[startMonth] + space + startHour + colon + endMinute;
+				endDate = endDay + space + months[endMonth] + space + endHour + colon + endMinute;
+			}
+		} else {
+			// same day - "dd MMM HH:MM - HH:MM"
+			startDate = startDay + space + months[startMonth] + space + startHour + colon + startMinute;
+			endDate = endHour + colon + endMinute;
+
+		}
+		date = startDate + dash + endDate;
+
+		//	location
+		location = eventDataExtended.location;
+
+		// description 
+		description = parseDescription(eventDataExtended.description);
+
+		// Facebook event button
+		let eventFacebookLinkElement = document.getElementById("facebook-link");
+		let link = parseFacebookLink(eventDataExtended.description);
+		if (link) {
+			eventFacebookLinkElement.onclick = () => {window.open(link)};
+			eventFacebookLinkElement.style.display = "inline";
+		} else {
+			// catches null regex matches
+			eventFacebookLinkElement.style.display = "none";
+		}
+	
+
+		// set text
+		document.getElementById("event-text-title").textContent = title;
+		document.getElementById("event-text-date").textContent = date;
+		document.getElementById("event-text-location").textContent = location;
+		var descStr = "";
+		for (line in description) {
+			descStr += description[line].toString() + "<br>";
+		}
+		document.getElementById("event-text-description").innerHTML = descStr; //innerhtml since description is already string
+		document.getElementById('calendar-event').style.display = "block"; //show
+	} else {
+		//null event
+		console.log("ERROR: event is null");
+		//should I keep this?
```

Comment:
yup keep this part. However, a special function exists for this, `console.error` which highlights in the console that it is an error, usually by highlighting in red.

Reply:
changed to `console.error`